app/assets/javascripts/channels/la_exercises.js

```diff
@@ -0,0 +1,250 @@
+                .attr("transform", function(d) {
+                    return "rotate(-45)"
+                });
+
+            svg.append("g")
+                .attr("class", "y axis")
+                .call(yAxis);
+
+            // Y Axis Label
+            svg.append("text")
+                .attr("transform", "rotate(-90)")
+                .attr("x", (-height - margin.top + margin.bottom) / 2)
+                .attr("dy", "+2em")
+                .style("text-anchor", "middle")
+                .text(I18n.t('exercises.study_group_dashboard.time_spent_in_minutes'))
+                .style('font-size', 14);
+
+            // X Axis Label
+            svg.append("text")
+                .attr("class", "x axis")
+                .attr("text-anchor", "middle")
+                .attr("x", (width + margin.left - margin.right) / 2)
+                .attr("y", height)
+                .attr("dy", '-1em')
+                .text(I18n.t('exercises.study_group_dashboard.learner'))
+                .style('font-size', 14);
+
+            let tip = d3.tip()
+                .attr('class', 'd3-tip')
+                .offset([-10, 0])
+                .html(function(d, i, a) {
+                    return "<strong>Student: </strong><span style='color:orange'>" + learners_name(i) + "</span><br/>" +
+                        "0: " + learners_time(0, i) + "<br/>" +
+                        "1: " + learners_time(1, i) + "<br/>" +
+                        "2: " + learners_time(2, i) + "<br/>" +
+                        "3: " + learners_time(3, i) + "<br/>" +
+                        "4: " + learners_time(4, i);
+                });
+
+            svg.call(tip);
+
+            rect.on('mouseenter', tip.show)
+                .on('mouseout', tip.hide);
+
+            function transitionGrouped() {
+                // Show all sub-bars next to each other
+                y.domain([0, maxYSingleBar]);
+
+                rect.transition()
+                    .duration(500)
+                    .delay((d, i) => i * 20)
+                    .attr("x", (d, i) => x(i) + x.bandwidth() / n * d[2])
+                    .attr("width", x.bandwidth() / n)
+                    .transition()
+                    .attr("y", d => y(d[1] - d[0]))
+                    .attr("height", d => y(0) - y(d[1] - d[0]));
+            }
+
+            function transitionStacked() {
+                // Show all sub-bars on top of each other
+                y.domain([0, maxYBarStacked]);
+
+                rect.transition()
+                    .duration(500)
+                    .delay((d, i) => i * 20)
+                    .attr("y", d => y(d[1]))
+                    .attr("height", d => y(d[0]) - y(d[1]))
+                    .transition()
+                    .attr("x", (d, i) => x(i))
+                    .attr("width", x.bandwidth());
+            }
+
+            $('#no_chart_data').addClass("d-none");
+            transitionStacked();
+            // transitionGrouped();
```

Comment:
is this a leftover?

Reply:
Somehow. I'll remove it here and add a comment to the method explaining what it does (the method `transitionGrouped()` is never called but might be interesting in the future. Or, I just add a button to the top right corner of the graph that allows switching between the stacked view and the grouped view.